e3sm_supported_machines/create_new_e3sm_unified_env.py

```diff
@@ -0,0 +1,259 @@
+            print('  {} failed'.format(exec))
+            raise e
+        print('  {} passes'.format(exec))
+
+    package = 'tempest-remap'
+    command = '{}; GenerateCSMesh --res 64 --alt --file ' \
+              'gravitySam.000000.3d.cubedSphere.g'.format(activate)
+    try:
+        subprocess.check_call(command, env=env, executable='/bin/bash',
+                              shell=True, stdout=subprocess.DEVNULL)
+    except subprocess.CalledProcessError as e:
+        print('  {} failed'.format(package))
+        raise e
+    print('  {} passes'.format(package))
+
+
+def get_host_info():
+    hostname = socket.gethostname()
+    if hostname.startswith('cori') or hostname.startswith('dtn'):
+        base_path = "/global/cfs/cdirs/e3sm/software/anaconda_envs/base"
+        activ_path = "/global/cfs/cdirs/e3sm/software/anaconda_envs"
+        group = "e3sm"
+    elif hostname.startswith('acme1') or hostname.startswith('aims4'):
+        base_path = "/usr/local/e3sm_unified/envs/base"
+        activ_path = "/usr/local/e3sm_unified/envs"
+        group = "climate"
+    elif hostname.startswith('blueslogin'):
+        base_path = "/lcrc/soft/climate/e3sm-unified/base"
+        activ_path = "/lcrc/soft/climate/e3sm-unified"
+        group = "climate"
+    elif hostname.startswith('rhea'):
+        base_path = "/ccs/proj/cli900/sw/rhea/e3sm-unified/base"
+        activ_path = "/ccs/proj/cli900/sw/rhea/e3sm-unified"
+        group = "cli900"
+    elif hostname.startswith('cooley'):
+        base_path = "/lus/theta-fs0/projects/ccsm/acme/tools/e3sm-unified/base"
+        activ_path = "/lus/theta-fs0/projects/ccsm/acme/tools/e3sm-unified"
+        group = "ccsm"
+    elif hostname.startswith('compy'):
+        base_path = "/share/apps/E3SM/conda_envs/base"
+        activ_path = "/share/apps/E3SM/conda_envs"
+        group = "users"
+    elif hostname.startswith('gr-fe') or hostname.startswith('wf-fe'):
+        base_path = "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/base"
+        activ_path = "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs"
+        group = "climate"
+    elif hostname.startswith('burnham'):
+        base_path = "/home/xylar/Desktop/test_e3sm_unified/base"
+        activ_path = "/home/xylar/Desktop/test_e3sm_unified"
+        group = "xylar"
+    else:
+        raise ValueError("Unknown host name {}.  Add env_path and group for "
+                         "this machine to the script.".format(hostname))
+
+    return base_path, activ_path, group
+
+
+def main():
+    # Modify the following list of dictionaries to choose which e3sm-unified
+    # version, python version, and which mpi variant (nompi, mpich or openmpi)
+    # to use.
+
+    envs = [{'suffix': '',
+             'version': '1.3.1',
+             'python': '3.7',
+             'mpi': 'nompi'},
+            {'suffix': '_mpich',
+             'version': '1.3.1',
+             'python': '3.7',
+             'mpi': 'mpich'}]
+
```

Comment:
You may want to put this in a config file (json?) that can be more easily edited. It's pretty simple now so might not be worth it, but would be if it grows in complexity. 

Altenatively, you might consider putting it at the top just under the imports like

```py
_ENVS = [...]
```
so you don't have to sroll down to line 109 to edit

Reply:
I **really** dislike module-level variables but I moved these into a little function that is now at the top of the script, so serves the same purpose.

Reply:
Also, I think a `json` is total overkill at this point but could be considered if the complexity really increases (in which case I'll probably already have abandoned E3SM-Unified in disgust).

Reply:
Looks good to me now! `get_envs` is a good idea; easy to change underlying functionality that way.